Sources/MessagingPush/Store/PushHistory.swift

```diff
@@ -0,0 +1,47 @@
+import CioInternalCommon
+import Foundation
+
+protocol PushHistory: AutoMockable {
+    func hasHandledPushClick(deliveryId: String) -> Bool
+    func handledPushClick(deliveryId: String)
+}
+
+// sourcery: InjectRegister = "PushHistory"
+class PushHistoryImpl: PushHistory {
+    private let keyValueStorage: KeyValueStorage
+
+    var numberOfPushesToTrack = 100
+
+    var lastPushesClicked: [String] {
+        get {
+            let stringRepresentationOfArray = keyValueStorage.string(.pushNotificationsClicked) ?? ""
+
+            return stringRepresentationOfArray.split(separator: ",").map { String($0) }
+        }
+        set {
+            let stringRepresentationOfArray = newValue.joined(separator: ",")
+            keyValueStorage.setString(stringRepresentationOfArray, forKey: .pushNotificationsClicked)
+        }
+    }
+
+    init(keyValueStorage: KeyValueStorage) {
+        self.keyValueStorage = keyValueStorage
+    }
+
+    func hasHandledPushClick(deliveryId: String) -> Bool {
+        lastPushesClicked.contains(deliveryId)
+    }
+
+    func handledPushClick(deliveryId: String) {
+        var clickHistory = lastPushesClicked
+
+        if clickHistory.count >= numberOfPushesToTrack {
+            // Remove oldest push click from history.
+            clickHistory = Array(clickHistory.dropFirst())
+        }
+
+        clickHistory.append(deliveryId)
+
+        lastPushesClicked = clickHistory
```

Comment:
Where can I find use case of this method `handledPushClick`? Is it in some other PR ?

Reply:
[It's used in this PR](https://github.com/customerio/customerio-ios/pull/405/files#diff-96954b50d5a6c450f707a82cae92b37ae877dfe98b15d2006d254cae04213a4dR123-R127)